src/NodaTime/Calendars/WeekYearRule.cs

```diff
@@ -0,0 +1,199 @@
+    /// a UK tax year rule could number weeks from April 6th onwards, such that any date earlier than that in the calendar
+    /// year would belong to the previous week-year.
+    /// </para>
+    /// <para>
+    /// The mapping of dates into week-year, week and day-of-week is always relative to a specific calendar system.
+    /// For example, years in the Hebrew calendar system vary very significantly in length due to leap months, and this
+    /// is reflected in the number of weeks within the week-years - as low as 50, and as high as 55.
+    /// </para>
+    /// <para>
+    /// This class allows conversions between the two schemes of identifying dates: <see cref="GetWeekYear(LocalDate)"/>
+    /// and <see cref="GetWeekOfWeekYear(LocalDate)"/> allow the week-year and week to be obtained, and
+    /// <see cref="CreateLocalDate(int, int, IsoDayOfWeek, CalendarSystem)"/> allows the reverse mapping. Note that
+    /// the calendar system does not need to be specified in the former methods as a <see cref="LocalDate"/> already
+    /// contains calendar information, and there is no method to obtain the day-of-week as that is not affected by the
+    /// week year rule being used.
+    /// </para>
+    /// <para>Specific rules are obtained using the static factory methods within this class, or with the
+    /// <see cref="Iso"/> property.</para>
+    /// <para>
+    /// All implementations within Noda Time are immutable, and it is advised that any external implementations
+    /// should be immutable too.
+    /// </para>
+    /// </remarks>
+    [Immutable]
+    public abstract class WeekYearRule
+    {
+        private static readonly WeekYearRule[] isoBasedRules = Enumerable.Range(1, 7).Select(x => new IsoBasedWeekYearRule(x)).ToArray();
+
+        /// <summary>
+        /// Returns a <see cref="WeekYearRule"/> similar to ISO-8601, but allowing the minimum number of
+        /// days in the first week to vary from the value of 4 specified by ISO-8601.
+        /// </summary>
+        /// <remarks>
+        /// <paramref name="minDaysInFirstWeek"/> determines when the first week of the week-year starts.
+        /// For any given calendar year X, consider the Monday-to-Sunday week that includes the first day of the
+        /// calendar year. Usually, some days of that week are in calendar year X, and some are in calendar year 
+        /// X-1. If <paramref name="minDaysInFirstWeek"/> or more of the days are in year X, then the week is
+        /// deemed to be the first week of week-year X. Otherwise, the week is deemed to be the last week of
+        /// week-year X-1, and the first week of week-year X starts on the following Monday.
+        /// </remarks>
+        /// <param name="minDaysInFirstWeek">The minimum number of days in the first Monday-to-Sunday week
+        /// which have to be in the new calendar year for that week to count as being in that week-year.
+        /// Must be in the range 1 to 7 inclusive.
+        /// </param>
```

Comment:
Not sure what you mean.... are you suggesting an extra sentence saying "If you want the ISO rules, just use the `Iso` property"?